### ISSUE TITLE
Phase 1.5: Build Basic Memory Storage and Retrieval System

### ISSUE THREAD

**Issue by RobeHGC** [RobeHGC/nadia-ai-companion#14]
## 🎯 Objective
Implement basic memory storage and retrieval system using Redis for short-term context and Neo4j for long-term episodic memories.

## 📦 Dependencies to Install
```bash
# Add to requirements.txt
sentence-transformers==2.2.2
transformers==4.35.0
torch==2.1.0
numpy==1.24.3
scikit-learn==1.3.0

# Install dependencies
pip install sentence-transformers transformers torch numpy scikit-learn
```

## 📋 Tasks
- [ ] Install required ML dependencies (sentence-transformers, torch)
- [ ] Implement Redis context window management (20-100 messages)
- [ ] Create Neo4j memory node storage system
- [ ] Build basic memory retrieval by similarity
- [ ] Implement memory embeddings with all-MiniLM-L6-v2
- [ ] Add memory metadata tracking (timestamp, user, session)
- [ ] Create memory cleanup and TTL management
- [ ] Test with sample conversations

## 📚 Documentation References
- `bitacora/memoria/MEMORIA.md` - Complete memory system design
- `bitacora/ARQUITECTURA_GLOBAL.md` - Lines 108-122 (Memory architecture)
- `bitacora/memoria/GUIA_COMPLETA.md` - Lines 110-133 (Memory schemas)
- `docker/neo4j/` - Database initialization scripts

## 🔧 Technical Implementation
- **Files**: `src/memoria/memory_manager.py`, `src/memoria/embeddings.py`
- **Redis**: Context window with user-specific keys
- **Neo4j**: Memory nodes with embeddings and metadata
- **Embeddings**: sentence-transformers/all-MiniLM-L6-v2 (384 dim)
- **Integration**: Connect to orchestrator and emotional analysis

## 🚀 Setup Instructions
1. Install dependencies: `pip install sentence-transformers transformers torch`
2. Initialize Neo4j with vector indexes: `python scripts/init_databases.py`
3. Test Redis connection: `python scripts/check_connections.py`
4. Run memory system tests: `pytest tests/test_memory.py`

## ✅ Definition of Done
- [ ] All dependencies installed and working
- [ ] Context window maintains configured message limit
- [ ] Memory nodes stored with proper schema in Neo4j
- [ ] Embedding generation works for all message types
- [ ] Memory retrieval returns relevant results
- [ ] TTL cleanup prevents memory bloat
- [ ] Performance meets latency requirements
- [ ] Integration tests pass

## 🗄️ Data Schema
- **Redis**: user:{id}:context - List of recent messages
- **Neo4j**: (:Memory) nodes with embedding, timestamp, emotional_weight
- **Relationships**: TEMPORALLY_RELATED, SEMANTICALLY_SIMILAR

## 📈 Success Metrics
- Memory retrieval accuracy > 75%
- Embedding generation time < 100ms
- Context window hit rate > 90%

**Comment by RobeHGC** [comment]
🎉 **Issue #14 COMPLETADO**

✅ **Todas las dependencias instaladas y funcionando:**
- PyTorch 2.7.1+cu126 ✅
- sentence-transformers 3.3.1 ✅ 
- transformers 4.45.2 ✅
- numpy 1.26.4 ✅
- scikit-learn 1.7.0 ✅

✅ **Servicios activos:**
- Redis ✅ (configurado y habilitado)
- Neo4j ✅ 
- PostgreSQL ✅

✅ **Sistema de memoria implementado:**
- Gestión de contexto Redis (20-100 mensajes)
- Almacenamiento de nodos en Neo4j con embeddings
- Recuperación por similitud con all-MiniLM-L6-v2
- Seguimiento de metadatos (timestamp, usuario, sesión)
- Limpieza TTL y gestión de memoria
- Integración con análisis emocional

**Commit:** 405dbe0 - feat: Complete Phase 1.5 - Memory Storage and Retrieval System (#14)

**Estado Final:** 95% completo - Solo faltan pruebas de integración
